### DISCUSSION TITLE
Serverless Function Size Issue

### DISCUSSION THREAD

**Discussion by ldgrs** [vercel/community#3355] [Help]
### Summary

Hello,

We are currently running our web application on vercel. The dev version is running on hobby and the live version is running on the paid package.

With increasing number of node modules installed, we are now beginning to have issues with Serverless Function Size Limit (50 compressed, 250 uncompressed). Though we already have deployed some of our runtime generated assets, e.g., images, svgs etc, on S3. 

The problem happens to be strictly with node js modules. Since we are also using node-canvas module, it is also taking up the major chunk of alloted limit.  

Can any one recommend any technique or method to be able to either handle or surpass that limitation? 

Any help in this regard will be much appreciated,

best regards,

### Steps to Reproduce

Hello,

We are currently running our web application on vercel. The dev version is running on hobby and the live version is running on the paid package.

With increasing number of node modules installed, we are now beginning to have issues with Serverless Function Size Limit (50 compressed, 250 uncompressed). Though we already have deployed some of our runtime generated assets, e.g., images, svgs etc, on S3. 

The problem happens to be strictly with node js modules. Since we are also using node-canvas module, it is also taking up the major chunk of alloted limit.  

Can any one recommend any technique or method to be able to either handle or surpass that limitation? 

Any help in this regard will be much appreciated,

best regards,

**Comment by amyegan** [comment]
I don't know the specifics of you app, and techniques vary based on project framework and architecture. You can find options for many different scenarios in this post: https://github.com/vercel/community/discussions/103
